pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -329,35 +349,94 @@ func (w *Webhook) getAllLatestLibraries() []libInfo {
 	return libsToInject
 }
 
-// extractLibInfo returns the language, the image,
-// and a boolean indicating whether the library should be injected into the pod
-func (w *Webhook) extractLibInfo(pod *corev1.Pod) ([]libInfo, bool) {
+type libInfoSource int
+
+const (
+	libInfoSourceNone libInfoSource = iota
+	libInfoSourceLibInjection
+	libInfoSourceSingleStepInstrumentation
+	libInfoSourceSingleStepLangaugeDetection
+)
+
+func (s libInfoSource) injectionType() string {
+	switch s {
+	case libInfoSourceSingleStepInstrumentation, libInfoSourceSingleStepLangaugeDetection:
+		return singleStepInstrumentationInstallType
+	case libInfoSourceLibInjection:
+		return localLibraryInstrumentationInstallType
+	default:
+		return "unknown"
+	}
+}
+
+func (s libInfoSource) autodetected() bool {
+	return s == libInfoSourceSingleStepLangaugeDetection
+}
+
+func (s libInfoSource) mutatePod(pod *corev1.Pod) error {
+	_ = mutatecommon.InjectEnv(pod, corev1.EnvVar{
+		Name:  instrumentationInstallTypeEnvVarName,
+		Value: s.injectionType(),
+	})
+	return nil
+}
+
+type extractedPodLibInfo struct {
+	// libs are the libraries we are going to attempt to inject into the given pod.
+	libs []libInfo
+	// languageDetection is set when we ran/used the language-detection annotation.
+	languageDetection *libInfoLanguageDetection
+	// source is where we got the data from, used for telemetry later.
+	source libInfoSource
+}
+
+func (e extractedPodLibInfo) withLibs(l []libInfo) extractedPodLibInfo {
+	e.libs = l
+	return e
+}
+
+// extractLibInfo metadata about what library information we should be
+// injecting into the pod and where it came from.
+func (w *Webhook) extractLibInfo(pod *corev1.Pod) extractedPodLibInfo {
+	var (
+		ssiEnabled        = w.isEnabledInNamespace(pod.Namespace)
+		langaugeDetection = w.getLibrariesLanguageDetection(pod)
+		extracted         = extractedPodLibInfo{
+			source:            libInfoSourceLibInjection,
+			languageDetection: &langaugeDetection,
+		}
+	)
+
+	if ssiEnabled {
+		extracted.source = libInfoSourceSingleStepInstrumentation
+	}
+
```

Comment:
if my understanding is correct, `ssiEnabled` will be `true` whener the namespace is enabled for APM SSI. 

It means that always in this case we will have `extracted.source = libInfoSourceSingleStepInstrumentation` no matter if the results were coming from pinned libraries or from process language detection.

Reply:
Yes, this is consistent with the old behavior (and there are tests for it). I can fix it here (I think this might be a bug) where if we're using the annotations we should be doing `libInjection`